packages/utilities/src/CliUtils.ts

```diff
@@ -95,6 +99,218 @@ export class CliUtils {
+                throw new ImperativeError({
+                    msg: `Profile of type "${profileType}" does not exist within the loaded profiles for the command and it is marked as required.`,
+                    additionalDetails: `This is an internal imperative error. ` +
+                        `Command preparation was attempting to extract option values from this profile.`
+                });
+            } else if (profile != null) {
+                // For each option - extract the value if that exact property exists
+                options.forEach((opt) => {
+
+                    // Get the camel an kebab case
+                    const cases = CliUtils.getOptionFormat(opt.name);
+
+                    // We have to "deal" with the situation that the profile contains both cases - camel and kebab.
+                    // This is to support where the profile options have "-", but the properties are camel case in the
+                    // yaml file - which is currently how most imperative CLIs have it coded.
+                    const profileKebab = profile[cases.kebabCase];
+                    const profileCamel = profile[cases.camelCase];
+
+                    // If the profile has either type (or both specified) we'll add it to args if the args object
+                    // does NOT already contain the value in any case
+                    if ((profileCamel !== undefined || profileKebab !== undefined) &&
+                        (!args.hasOwnProperty(cases.kebabCase) && !args.hasOwnProperty(cases.camelCase))) {
+
+                        // If both case properties are present in the profile, use the one that matches
+                        // the option name explicitly
+                        const value = (profileKebab !== undefined && profileCamel !== undefined) ?
+                            ((opt.name === cases.kebabCase) ? profileKebab : profileCamel) :
+                            ((profileKebab !== undefined) ? profileKebab : profileCamel);
+                        const keys = CliUtils.setOptionValue(opt.name,
+                            ("aliases" in opt) ? (opt as ICommandOptionDefinition).aliases : [],
+                            value
+                        );
+                        args = {...args, ...keys};
+                    }
+                });
+            }
+        });
+        return args;
+    }
+
+    /**
+     * Using Object.assign(), merges objects in the order they appear in call. Object.assign() copies and overwrites
+     * existing properties in the target object, meaning property precedence is least to most (left to right).
+     *
+     * See details on Object.assign() for nuance.
+     *
+     * @param {...any[]} args - variadic set of objects to be merged
+     *
+     * @returns {*} - the merged object
+     *
+     */
+    public static mergeArguments(...args: any[]): any {
+        let merged = {};
+        args.forEach((obj) => {
+            merged = {...merged, ...obj};
```

Comment:
:warning: I hope you guys are aware that this is not a deep copy of the objects.

Reply:
Yes this is OK. It's really only keyword/value pairs.

Reply:
Thanks for clarifying